diff --git a/megatron/data/indexed_dataset.py b/megatron/data/indexed_dataset.py
index 3b4f822..03a696b 100644
--- a/megatron/data/indexed_dataset.py
+++ b/megatron/data/indexed_dataset.py
@@ -95,7 +95,7 @@ dtypes = {
     3: np.int16,
     4: np.int32,
     5: np.int64,
-    6: np.float,
+    6: float,
     7: np.double,
     8: np.uint16
 }
@@ -268,7 +268,7 @@ class IndexedDatasetBuilder(object):
         np.int16: 2,
         np.int32: 4,
         np.int64: 8,
-        np.float: 4,
+        float: 4,
         np.double: 8
     }
 
diff --git a/megatron/fused_kernels/fused_weight_gradient_dense.cu b/megatron/fused_kernels/fused_weight_gradient_dense.cu
index 7dc10e6..b9ca3ef 100644
--- a/megatron/fused_kernels/fused_weight_gradient_dense.cu
+++ b/megatron/fused_kernels/fused_weight_gradient_dense.cu
@@ -45,7 +45,7 @@ cublasStatus_t gemmex_wrapper(
       C,
       CUDA_R_32F,
       ldc,
-      CUDA_R_32F,
+      CUBLAS_COMPUTE_32F,
       CUBLAS_GEMM_DEFAULT_TENSOR_OP);
 }
 
@@ -83,7 +83,7 @@ cublasStatus_t gemmex_wrapper(
       C,
       CUDA_R_32F,
       ldc,
-      CUDA_R_32F,
+      CUBLAS_COMPUTE_32F,
       CUBLAS_GEMM_DEFAULT_TENSOR_OP);
 }
 
@@ -121,7 +121,7 @@ cublasStatus_t gemmex_wrapper(
       C,
       CUDA_R_32F,
       ldc,
-      CUDA_R_32F,
+      CUBLAS_COMPUTE_32F,
       CUBLAS_GEMM_DEFAULT_TENSOR_OP);
 }
 
diff --git a/megatron/fused_kernels/scaled_masked_softmax_cuda.cu b/megatron/fused_kernels/scaled_masked_softmax_cuda.cu
index a8be57c..674db48 100644
--- a/megatron/fused_kernels/scaled_masked_softmax_cuda.cu
+++ b/megatron/fused_kernels/scaled_masked_softmax_cuda.cu
@@ -4,7 +4,6 @@
 #include <cuda.h>
 #include <cuda_runtime.h>
 #include <cuda_fp16.h>
-#include <cuda_profiler_api.h>
 #include <ATen/cuda/CUDAContext.h>
 #include <torch/extension.h>
 #include "scaled_masked_softmax.h"
diff --git a/megatron/fused_kernels/scaled_softmax_cuda.cu b/megatron/fused_kernels/scaled_softmax_cuda.cu
index ecc6eb0..b536065 100644
--- a/megatron/fused_kernels/scaled_softmax_cuda.cu
+++ b/megatron/fused_kernels/scaled_softmax_cuda.cu
@@ -4,7 +4,6 @@
 #include <cuda.h>
 #include <cuda_runtime.h>
 #include <cuda_fp16.h>
-#include <cuda_profiler_api.h>
 #include <ATen/cuda/CUDAContext.h>
 #include <torch/extension.h>
 #include "scaled_masked_softmax.h"
diff --git a/megatron/fused_kernels/scaled_upper_triang_masked_softmax_cuda.cu b/megatron/fused_kernels/scaled_upper_triang_masked_softmax_cuda.cu
index c21e5fb..85e6162 100644
--- a/megatron/fused_kernels/scaled_upper_triang_masked_softmax_cuda.cu
+++ b/megatron/fused_kernels/scaled_upper_triang_masked_softmax_cuda.cu
@@ -4,7 +4,6 @@
 #include <cuda.h>
 #include <cuda_runtime.h>
 #include <cuda_fp16.h>
-#include <cuda_profiler_api.h>
 #include <ATen/cuda/CUDAContext.h>
 #include <torch/extension.h>
 #include "scaled_upper_triang_masked_softmax.h"
diff --git a/megatron/initialize.py b/megatron/initialize.py
index c10b7bf..366803f 100644
--- a/megatron/initialize.py
+++ b/megatron/initialize.py
@@ -245,7 +245,7 @@ def set_jit_fusion_options():
         torch._C._jit_override_can_fuse_on_cpu(False)
         torch._C._jit_override_can_fuse_on_gpu(False)
         torch._C._jit_set_texpr_fuser_enabled(False)
-        torch._C._jit_set_nvfuser_enabled(True)
+        torch._C._jit_set_nvfuser_enabled(False)
         torch._C._debug_set_autodiff_subgraph_inlining(False)
     else:
         # legacy pytorch fuser
diff --git a/megatron/model/fused_layer_norm.py b/megatron/model/fused_layer_norm.py
index a7115b0..28b0917 100644
--- a/megatron/model/fused_layer_norm.py
+++ b/megatron/model/fused_layer_norm.py
@@ -26,16 +26,20 @@ fused_layer_norm_cuda = None
 class FusedLayerNormAffineFunction(torch.autograd.Function):
 
   @staticmethod
-  def forward(ctx, input, weight, bias, normalized_shape, eps):
+  def forward(ctx, input, weight, bias, normalized_shape, eps, memory_efficient=False):
 
     ctx.normalized_shape = normalized_shape
     ctx.eps = eps
+    ctx.memory_efficient = memory_efficient
     input_ = input.contiguous()
     weight_ = weight.contiguous()
     bias_ = bias.contiguous()
     output, mean, invvar = fused_layer_norm_cuda.forward_affine(
         input_, ctx.normalized_shape, weight_, bias_, ctx.eps)
-    ctx.save_for_backward(input_, weight_, bias_, mean, invvar)
+    if ctx.memory_efficient:
+        ctx.save_for_backward(input_, weight_, bias_, None, invvar)
+    else:
+        ctx.save_for_backward(output, weight_, bias_, mean, invvar)
 
     return output
 
@@ -49,7 +53,7 @@ class FusedLayerNormAffineFunction(torch.autograd.Function):
       = fused_layer_norm_cuda.backward_affine(
         grad_output.contiguous(), mean, invvar,
         input_, ctx.normalized_shape,
-        weight_, bias_, ctx.eps)
+        weight_, bias_, ctx.eps, ctx.memory_efficient)
 
     return grad_input, grad_weight, grad_bias, None, None
 
